app.py

```diff
@@ -714,23 +714,38 @@ def redraw():
 	"""Validate and re-draw tree."""
 	data = request.get_json()
 	sentno = int(data.get('sentno')) # 1-indexed
-	treeobj = ActivedopTree.from_str(data.get('tree'))
+	has_error = False
+	link = ('''<a href="#" onclick="accept()">accept this tree</a>
+		<input type="hidden" id="sentno" value="%d">'''
+	% (sentno))
+	try:
+		treeobj = ActivedopTree.from_str(data.get('tree'))
+		msg = treeobj.validate()
+	except Exception as err:
+		if len(str(err)) > 0:
+			error_msg = str(err)
+		else:
+			error_msg = "Your CGEL tree is structurally deficient."
```

Comment:
This sounds like it could be a grammatical syntax issue. How about: "Format error in CGEL input. Check for matching parentheses, quotes, etc."

Reply:
Updated in https://github.com/nschneid/activedop/commit/5b0f2980caa8fbb0ad2e4f997582707253b47c3f